### PR TITLE
Test/unobs lc yield

### PR DESCRIPTION
**Pull request by MishkaMN**
# PR Details
## Description
This PR contains only the necessary components for cooperative lanechange functionality that only relates to yield plugin. 
 

## Related Issue
https://github.com/usdot-fhwa-stol/carma-platform/issues/1227

## Motivation and Context
See above.

## How Has This Been Tested?
Integration tested at Summit Point Fusion

## Types of changes

- [X] Defect fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that cause existing functionality to change)

## Checklist:

- [X] I have added any new packages to the sonar-scanner.properties file
- [ ] My change requires a change to the documentation.
- [X] I have updated the documentation accordingly.
- [X] I have read the **CONTRIBUTING** document.
[CARMA Contributing Guide](https://github.com/usdot-fhwa-stol/carma-platform/blob/develop/Contributing.md) 
- [X] I have added tests to cover my changes.
- [X] All new and existing tests passed.

### CODE REVIEW

File: inlanecruising_plugin/src/inlanecruising_plugin.cpp [hunk 1]
```diff
@@ -701,7 +704,7 @@ bool InLaneCruisingPlugin::validate_yield_plan(const cav_msgs::TrajectoryPlan& y
   {
     ROS_DEBUG_STREAM("Invalid Yield Trajectory"); 
   }
-  return false;
+  return true;
```
Comment (msmcconnell) [inline inlanecruising_plugin/src/inlanecruising_plugin.cpp:707]: The line right above this states this is an invalid plan and yet we are returning true???

File: yield_plugin/src/yield_plugin.cpp [hunk 2]
```diff
@@ -170,22 +192,26 @@ namespace yield_plugin
...
+        ROS_DEBUG_STREAM("req_traj_speed" << req_traj_speed);
+        ROS_DEBUG_STREAM("start_lanelet_id" << start_lanelet_id);
+        ROS_DEBUG_STREAM("end_lanelet_id" << end_lanelet_id);
 
         std::vector<lanelet::BasicPoint2d> req_traj_plan = {};
 
         req_traj_plan = convert_eceftrajectory_to_mappoints(incoming_trajectory, tf_);
 
-        double req_expiration_sec = incoming_request.expiration/1000;
+        double req_expiration_sec = (double)incoming_request.expiration; //ms to second
```
Comment (msmcconnell) [inline yield_plugin/src/yield_plugin.cpp:208]: What part of this line is ms/seconds???

Comment (MishkaMN) [reply yield_plugin/src/yield_plugin.cpp:208]: Ah that is bad comment thanks

File: inlanecruising_plugin/src/inlanecruising_plugin.cpp (hunk 1 above)
Comment (kjrush) [reply inlanecruising_plugin/src/inlanecruising_plugin.cpp:707]: Agreed, not sure I understand the intent of this change, it appears that in the original code this was a fallthrough case to return false, which makes sense.

File: mobilitypath_publisher/src/mobilitypath_publisher.cpp [hunk 3]
```diff
@@ -131,10 +131,6 @@ namespace mobilitypath_publisher
                 traj.offsets.push_back(offset);
             }
         }
-        //according to msg standard, ecef_location is in meters
```
Comment (kjrush) [inline mobilitypath_publisher/src/mobilitypath_publisher.cpp:134]: Is the message standard wrong or was this comment incorrect before?

File: mobilitypath_visualizer/src/mobilitypath_visualizer.cpp [hunk 4]
```diff
@@ -134,11 +134,6 @@ namespace mobilitypath_visualizer {
 
         auto curr_location_msg = msg; //variable to update on each iteration as offsets are measured since last traj point
         
-        //location message standard is specified in meters, while the logic here expects cm to work with offsets
```
Comment (kjrush) [inline mobilitypath_visualizer/src/mobilitypath_visualizer.cpp:137]: Same as above, one or the other should be updated if this is no longer needed.

File: yield_plugin/config/parameters.yaml [hunk 5]
```diff
@@ -47,11 +47,11 @@ enable_cooperative_behavior: true
 always_accept_mobility_request: true
 # Acceptable number of timesteps to use the latest known mobility request before switching to yield
 # Value type: Desired
-acceptable_passed_timesteps: 3
+acceptable_passed_timesteps: 10
```
Comment (kjrush) [inline yield_plugin/config/parameters.yaml:50]: A whole second seems like a lot, but if it's been working in the field I suppose it's best not to play around with it.

File: yield_plugin/include/yield_plugin/yield_plugin_node.h [hunk 6]
```diff
@@ -77,7 +77,7 @@ class YieldPluginNode
     
     worker.set_lanechange_status_publisher(lc_status_pub);
 
-    ros::ServiceServer trajectory_srv_ = nh.advertiseService("plugins/Yieldlugin/plan_trajectory",
+    ros::ServiceServer trajectory_srv_ = nh.advertiseService("plugins/YieldPlugin/plan_trajectory",
```
Comment (kjrush) [inline yield_plugin/include/yield_plugin/yield_plugin_node.h:80]: Ah, good catch. This is the kind of thing that hopefully refactoring this boilerplate into a base class can solve for us, that these service creations/etc can be automatically defined for us.

File: yield_plugin/src/yield_plugin.cpp [hunk 7]
```diff
@@ -248,11 +277,13 @@ namespace yield_plugin
     {
       if (timesteps_since_last_req_ < config_.acceptable_passed_timesteps)
       {
+        ROS_DEBUG_STREAM("Yield for CLC");
         yield_trajectory = update_traj_for_cooperative_behavior(original_trajectory, req.vehicle_state.longitudinal_vel);
-        timesteps_since_last_req_++;
+        // timesteps_since_last_req_++;
```
Comment (kjrush) [inline yield_plugin/src/yield_plugin.cpp:282]: So are we never expiring the yield plan if we're not incrementing this?

Comment (kjrush) [reply yield_plugin/src/yield_plugin.cpp:282]: The log statement here should probably also note that we haven't received an updated negotiation this timestep, so it's obvious in post-analysis.

File: mobilitypath_publisher/src/mobilitypath_publisher.cpp (hunk 3 above)
Comment (MishkaMN) [reply mobilitypath_publisher/src/mobilitypath_publisher.cpp:134]: the comment and this /100 logic was incorrect before. So I updated it to use cm, so we don't have this code. Rest of the packages also use this metrics

File: mobilitypath_visualizer/src/mobilitypath_visualizer.cpp (hunk 4 above)
Comment (MishkaMN) [reply mobilitypath_visualizer/src/mobilitypath_visualizer.cpp:137]: Same as above

File: yield_plugin/config/parameters.yaml (hunk 5 above)
Comment (saina-ramyar) [reply yield_plugin/config/parameters.yaml:50]: back to 3, it is working well with both values.

File: yield_plugin/src/yield_plugin.cpp (hunk 7 above)
Comment (saina-ramyar) [reply yield_plugin/src/yield_plugin.cpp]: removed to test its impact. It is back now.

**Review by kjrush — APPROVED** [review]

**Pull request merged**